pkg/secrets/check_rights_nix.go

```diff
@@ -13,15 +13,23 @@ import (
 	"syscall"
 )
 
-func checkRights(path string) error {
+func checkRights(path string, allowGroupExec bool) error {
 	var stat syscall.Stat_t
 	if err := syscall.Stat(path, &stat); err != nil {
 		return fmt.Errorf("invalid executable '%s': can't stat it: %s", path, err)
 	}
 
 	// checking that group and others don't have any rights
-	if stat.Mode&(syscall.S_IRWXG|syscall.S_IRWXO) != 0 {
-		return fmt.Errorf("invalid executable '%s', 'groups' or 'others' have rights on it", path)
+	if allowGroupExec {
+		// Allow group to have read and exec rights, blacklisting write
+		if stat.Mode&(syscall.S_IRWXG|syscall.S_IWGRP) != 0 {
```

Comment:
I'm not sure this is what you meant. Did you mean this instead?

`stat.Mode&(syscall.S_IWGRP|syscall.S_IRWXO) != 0`

Reply:
good catch, 
I messed up the 2 bit masks in the `if else`. Should be good now.

Reply:
Now this bug is back after your force-push...